proposals/types-architecture.md

```diff
@@ -0,0 +1,68 @@
+# Types architecture
+
+The hypothetical problem is thus: we have a global `FestivalModel` that represents a normalised festival with all fields populated. We also have a `Location` component that is a tiny section on the `FestivalCard`. 
+
+How should we organise our types?
+
+## Option 1 
+
+Types are redeclared in the component. This will result in duplication, but conceptually this approach is more encapsulated. When it comes to 'dumb' components, `Location` doesn't need to be aware of the whole FestivalModel. It just displays two strings.
+
+```jsx
+// app/types/festivals/index.ts
+export interface FestivalModel {
+  location: {
+    x: string;
+    y: string;
+  }
+}
+
+// app-components/components/festivals/FestivalCard/components/Location/index.tsx
+interface LocationProps {
+  x: string;
+  y: string;
+}
+
+export function Location({ x, y }: LocationProps) {
+  return <div>{x}, {y}</div>
+}
+```
+
+## Option 2
+`Location` accesses the type from the `FestivalModel`. This prevents the need for duplication and ensures consistent typing, but it also makes our components less 'dumb' as they know about the `FestivalModel`.
+
+```jsx
+// app/types/festivals/index.ts
+export interface FestivalModel {
+  location: {
+    x: string;
+    y: string;
+  }
+}
+
+// app-components/components/festivals/FestivalCard/components/Location/index.tsx
+export function Location({ x, y }: FestivalModel['location']) {
+  return <div>{x}, {y}</div>
+}
+```
+
+## Option 3
+
+Our model type is composed of other types. This means we can readily import the `FestivalLocation` type in `Location`. Again, we avoid duplication and ensure consistency but the 'dumb' component becomes less dumb. Also, our generic types file becomes very bloated.
```

Comment:
not sure I would agree it about bloat, because in Location is pretty reusable. Accommodation types would probably use it also. I prefer this option.

I think we could probably get a lot of value from talking to backend about stuff like this? I assume on the backend a location is always a location no matter what it's attached to

Reply:
**I would personally go for 3rd option**
+ Easier to reuse type Location, then using notation Festival[‘location’], IDE will auto-complete it when typed, plus more clear type definitions
+ We provide more flexibility as location can be reused in other type

**1rst option**
Encapsulation - should be used for isolating component related props but in this case, it is not purely component related structure, it comes from outside. The downside then is duplication of type which might cause errors. Ex even though we use copy&paste, we can forget some places and will not be warned if location object in general has changed, and component logic should be updated (ex selector returns new location type and it is reused in different components, that have individual location interface described)

**2 option**
The biggest drawback for me would be not having IDE support to get the type plus less clear type definition. The second is  reusability. Ex: take Availability type into example.

Reply:
I prefer the 3rd option too, but for similar reasons as @sanpoChew and @rlengvenis . 
Don't have anything more to add here

Reply:
Option 3 for me too